src/app/phast/phast.service.ts

```diff
@@ -891,5 +907,291 @@ export class PhastService {
+      phast.losses.wallLosses.forEach(loss => {
+        let wallForm: FormGroup = this.wallLossesService.getWallLossForm(loss);
+        if (wallForm.status === 'INVALID') {
+          valid = false;
+        }
+      });
+    }
+    return valid;
+  }
+
+  checkCoolingValid(phast: PHAST): boolean {
+    let valid = true;
+    if (phast.losses.coolingLosses) {
+      let coolingForm: FormGroup;
+      phast.losses.coolingLosses.forEach(loss => {
+        if (loss.gasCoolingLoss) {
+          coolingForm = this.coolingLossesService.initGasFormFromLoss(loss);
+        } else if (loss.liquidCoolingLoss) {
+          coolingForm = this.coolingLossesService.initLiquidFormFromLoss(loss);
+        }
+
+        if (coolingForm.status === 'INVALID') {
+          valid = false;
+        }
+      });
+    }
+    return valid;
+  }
+
+  checkAtmosphereValid(phast: PHAST): boolean {
+    let valid = true;
+    if (phast.losses.atmosphereLosses) {
+      phast.losses.atmosphereLosses.forEach(loss => {
+        let atmosphereForm: FormGroup = this.atmosphereLossesService.getAtmosphereForm(loss)
+        if (atmosphereForm.status === 'INVALID') {
+          valid = false;
+        }
+      });
+    }
+    return valid;
+  }
+
+  checkOpeningValid(phast: PHAST): boolean {
+    let valid = true;
+    if (phast.losses.openingLosses) {
+      phast.losses.openingLosses.forEach(loss => {
+          let openingForm: FormGroup = this.openingLossesService.getFormFromLoss(loss);
+          if (openingForm.status === 'INVALID') {
+            valid = false;
+          }
+        });
+    }
+    return valid;
+  }
+
+  checkLeakageValid(phast: PHAST): boolean {
+    let valid = true;
+    if (phast.losses.leakageLosses) {
+      phast.losses.leakageLosses.forEach(loss => {
+        let leakageForm: FormGroup = this.gasLeakageLossesService.initFormFromLoss(loss)
+        if (leakageForm.status === 'INVALID') {
+          valid = false;
+        }
+      });
+    }
+    return valid;
+  }
+
+  checkExtendedValid(phast: PHAST): boolean {
+    let valid = true;
+    if (phast.losses.extendedSurfaces) {
+      phast.losses.extendedSurfaces.forEach(loss => {
+        // Pass in wall loss here or exetended surface?
+        let extendedSurfaceForm: FormGroup = this.extendedSurfaceLossesService.getSurfaceLossForm(loss)
+        if (extendedSurfaceForm.status === 'INVALID') {
+          valid = false;
+        }
+      });
+    }
+    return valid;
+  }
+
+  checkOtherValid(phast: PHAST): boolean {
+    let valid = true;
+    if (phast.losses.otherLosses) {
+      phast.losses.otherLosses.forEach(loss => {
+        let otherLossForm: FormGroup = this.otherLossessService.getFormFromLoss(loss)
+        if (otherLossForm.status === 'INVALID') {
+          valid = false;
+        }
+      });
+    }
+    return valid;
+  }
+
+  checkSysEfficiencyValid(phast: PHAST): boolean {
+    const efficiencyValue = this.heatSystemEfficiencyCompareService.modification.systemEfficiency;
```

Comment:
"let"

should be `phast.systemEfficiency` 

using compare service will only ever check the selected modification which is not the desired behavior

Reply:
Others look good, guessing copy/paste thing